pages/functional/js/fizz-buzz.js

```diff
@@ -1,55 +1,51 @@
 import { parseNumber } from "../../js/helper.js";
 
 /**
- * Create an array filled by "fizz", "buzz", or "fizz buzz" based on
- * this requirement below (highest priority from above).
- *
- * Muliples of 4 or 7: "fizz buzz"
- * Odd sequence: "fizz"
- * Even sequence: "buzz"
- *
- * @param {number} sequence
- */
-function generateFizzBuzz(sequence) {
-  sequence = parseNumber(sequence);
-
-  let result = [];
-
-  for (let index = 1; index <= sequence; index++) {
-    if (index % 4 === 0 || index % 7 === 0) {
-      result.push("fizz buzz");
-    } else if (index % 2 === 1) {
-      result.push("fizz");
-    } else if (index % 2 === 0) {
-      result.push("buzz");
-    }
+* Using Simple Loop
+* @param {number} sequence
+*/
+function generateFizzBuzz (sequence) {
+  for (let i = 1; i <= sequence; i++) {
+      if (i % 4 === 0 || i % 7 === 0) {
+          console.log (`fizz buzz ${i}`);
+          continue;
+      } else if (i % 2 === 1) {
+          console.log (`fizz ${i}`)
+      } else if (i % 2 === 0) {
+          console.log (`buzz ${i}`)
+      } 
   }
-
-  return result;
 }
 
-document.getElementById("form").addEventListener("submit", function (event) {
-  event.preventDefault();
+generateFizzBuzz (8);
 
-  try {
-    const sequence = event.target["sequence"].value;
+/**
+* Using concept "Const of" and  "Function as a parameter"
```

Comment:
The usage of `for of` is correct, but the `function as a parameter` is wrong. If you want to pass the `numberGenerator` function as a parameter for the other function (ex: generateFizzBuzz2), don't add the paranthesis at the end of `numberGenerator`. See this example below:
```js
/**
 * @param {function} callback
 */
function generateFizzBuzz2(callback) {
  callback(8);
}

/**
 * @param {number} value
 */
const numberGenerator = function (value) {
  return value;
}

generateFizzBuzz2(numberGenerator);
```